### PR TITLE
Migrate embedded gateway step

### PR DESCRIPTION
**Pull request by pihme**
## Description

Migrates embedded gateway step.

## Related issues
#7539

## Review Hints
* There will be another PR which looks at all tests and removes public keywords and such

## Definition of Done

Code changes:
* [X] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/zeebe-io/zeebe/compare/stable/0.24...develop?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/0.25`) to the PR, in case that fails you need to create backports manually.

Testing:
* [X] There are unit/integration tests that verify all acceptance criterias of the issue
* [X] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark 

Documentation: 
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)

### CODE REVIEW

File: broker/src/test/java/io/camunda/zeebe/broker/bootstrap/EmbeddedGatewayServiceStepTest.java
Comment (npepinpe) [inline broker/src/test/java/io/camunda/zeebe/broker/bootstrap/EmbeddedGatewayServiceStepTest.java:96]: :wrench: IIRC, AssertJ has future specific assertions, so you can do things like:

```java
assertThat(startupFuture).succeedsWithin(Duration.ofSeconds(10)).isEqualTo(testBrokerStartupContext);
```

Or when you expect it to fail:

```java
assertThat(startupFuture).failsWithin(Duration.ofSeconds(5)).isInstanceOf(ExecutionException.class).havingRootCause(NullPointerException.class);
```

**Review by npepinpe — APPROVED** [review]
Generally looks good, had one thing but no need for another review for it, and some smaller suggestions. Please fix (or challenge :slightly_smiling_face:) the blocker before merging.

File: broker/src/test/java/io/camunda/zeebe/broker/bootstrap/EmbeddedGatewayServiceStepTest.java
Comment (pihme) [reply broker/src/test/java/io/camunda/zeebe/broker/bootstrap/EmbeddedGatewayServiceStepTest.java:96]: I will do this in a general code cleanup for all tests

**Comment by pihme** [comment]
bors merge

**Pull request merged**
